index.bs

```diff
@@ -795,6 +741,528 @@ the following list:
 
 </div>
 
+
+
+# The {{WebSocketStream}} interface # {#the-websocketstream-interface}
+
+The Web IDL definition for the {{WebSocketStream}} class is given as follows:
+
+<xmp class="idl">
+dictionary WebSocketOpenInfo {
+  ReadableStream readable;
+  WritableStream writable;
+  DOMString extensions;
+  DOMString protocol;
+};
+
+dictionary WebSocketCloseInfo {
+  [Clamp] unsigned short closeCode;
+  USVString reason = "";
+};
+
+dictionary WebSocketStreamOptions {
+  sequence<USVString> protocols;
+  AbortSignal signal;
+};
+
+[
+    Exposed=(Window,Worker)
+] interface WebSocketStream {
+  constructor(USVString url, optional WebSocketStreamOptions options);
+  readonly attribute USVString url;
+  readonly attribute Promise<WebSocketOpenInfo> opened;
+  readonly attribute Promise<WebSocketCloseInfo> closed;
+  undefined close(optional WebSocketCloseInfo closeInfo);
+};
+</xmp>
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">url</dfn> (a [=URL
+record=]).
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">opened promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">closed promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">readable stream</dfn>,
+which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">writable stream</dfn>,
+which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">was ever connected
+flag</dfn>, which is initially unset.
+
+A {{WebSocketStream}} object has an associated <dfn for="WebSocketStream">ready state</dfn>, which
+is a number representing the state of the connection. Initially it must be {{WebSocket/CONNECTING}}
+(0). It has the same semantics as {{WebSocket}}'s [=WebSocket/ready state=], but is not exposed to
+JavaScript.
+
+
+<dl class="domintro non-normative">
+ : <code>|socket| = new {{WebSocketStream/constructor(url, options)|WebSocketStream}}(|url| [, |options| ]</code>
+ :: Creates a new {{WebSocketStream}} object, immediately establishing the associating WebSocket
+  connection.
+
+  |url| is a string giving the <a dfn spec=url>URL</a> over which the connection is established.
+  Only "`ws`" or "`wss`" schemes are allowed; others will cause a "{{SyntaxError}}"
+  {{DOMException}}. URLs with [=fragments=] will also cause such an exception.
```

Comment:
We should allow HTTP(S) here. In fact, perhaps those are the only ones we should allow given it's a new entry point?

Reply:
Ah, sorry. I wrote this bit before that change was made. The actual algorithm steps are correct.

Reply:
It's fixed now.